openpype/lib/python_module_tools.py

```diff
@@ -129,20 +130,14 @@ def classes_from_module(superclass, module):
     for name in dir(module):
         # It could be anything at this point
         obj = getattr(module, name)
-        if not inspect.isclass(obj):
-            continue
-
-        # These are subclassed from nothing, not even `object`
-        if not len(obj.__bases__) > 0:
+        if not inspect.isclass(obj) or obj is superclass:
             continue
 
         # Use string comparison rather than `issubclass`
         # in order to support reloading of this module.
```

Comment:
The code below has changed ignoring this comment. So either this comment is incorrect and the original 'issue' is resolved and this comment can removed? Or we reintroduced an old issue?

Reply:
Now you scared me... I've read the comment 5 times and still don't get how I meant it?

Reply:
Basically the comment states that it is _not_ using `issubclass` for a reason but you have changed the lines below it to use `issubclass`. :) So now the comment makes no sense with the code.

Basically you deviated from exactly what the comment was trying to explain not to do. Either remove the comment and "disallow it still working when reloading that module" OR we preserve the old behavior.

Reply:
I've look in history and it was added with `LaunchHooks` but why would using class name instead of issubclass checks help to reload module? Nonsense.

Reply:
Say you have a file `a` defining class `Plugin`. You have other files, e.g. file `b`, importing it and inheriting from `Plugin`.

Once you reload file `a` then the plugin defined in `b` isn't a subclass (until you'd afterwards also reload `b`) anymore due to it inheriting from the original class before the reload. However by name they'd still match.

With how discover is implemented and thus plugins always reloading this might not seem like a direct issue.. but you would still hit the issue if e.g. plugin file `b` used an intermediate class type, like the Maya ReferenceLoader or anything from a file that doesn't get as dynamically loaded then the issue would be pretty apparent.

I believe however the issue would only occur if you'd reload file `a`. Or to be more descriptive, if you would reload the files from where the Plugin superclasses are loaded.